pgosupport.cmake

```diff
@@ -19,25 +28,57 @@ function(add_pgo TargetName)
     )
 
     # Enable PGO only for optimized configs
-    set(ConfigTypeList RELEASE RELWITHDEBINFO)
+    set(ConfigList RELEASE RELWITHDEBINFO)
+    set(IsReleaseConfig "$<OR:$<CONFIG:RELEASE>,$<CONFIG:RELWITHDEBINFO>>")
 
-    foreach(ConfigType IN LISTS ConfigTypeList)
-        set(LinkFlagsProperty "LINK_FLAGS_${ConfigType}")
-        if(CLR_CMAKE_PGO_INSTRUMENT)
+    if(CLR_CMAKE_PGO_INSTRUMENT)
+        # Unfortunately LINK_FLAGS_* don't support generator expressions, so we need to use a loop
+        foreach(Config IN LISTS ConfigList)
```

Comment:
Sure, I'm happy to rewrite it without the loop and without the helper function to keep it easier to follow.

Problem with the suggestion for LINK_FLAGS is that even on Unix, we need the flags applied only when configuring for release or relwithdebinfo. If I add it to LINK_FLAGS, they would get applied even during a debug build. Generator expressions would be the go-to answer, but they're not supported for LINK_FLAGS or its variants.

I did some private testing and it seems like cmake does the right thing with LINK_FLAGS_{CONFIG}, even when targeting a single-config build system like `make`. The only other approach I can think of is to use target_link_libraries() to add linker flags instead--which *do* support generator expressions--but they end up putting the flags inside of the "additional libraries" msbuild property, which I didn't like the sight of (it works, but seems like more of a hack to me).